Change module API + QA improvements

rexxars opened the pull request:
This PR changes the module API to accept an object instead of individual parameters, as discussed in #10.

I also took the liberty of adding tests for the CLI tool, refactored some code to minimize code duplication and replaced the aging JSHint configuration with ESLint. In the CLI tests, I've used `spawnSync`, which is only available in Node >= 0.12. I think it's fine to encourage people to upgrade, so I've set the minimum version in package.json to match, but the tests could be refactored if we think 0.10 support is still important.

Obviously this should be released as 2.0 since this introduces breaking changes. To make it easier to upgrade, I've added a changelog.

Sorry about the big PR, I probably should have split this one into smaller chunks.

rstacruz commented on Gruntfile.js:
```diff
@@ -1,19 +0,0 @@
-
-    // Default task.
-    grunt.registerTask('default', 'jshint');
-};
```
thank you for removing grunt! it's silly to be using grunt just for jshint imho

rstacruz commented on README.md:
```diff
@@ -70,13 +70,23 @@ travis-encrypt --add -r pwmckenna/node-travis-encrypt ENV1=VALUE1 ENV2=VALUE2
+  data: 'EXAMPLE_ENV_VARIABLE=asdf',
+  username: 'username',
+  password: 'password',
+  function (err, blob) {
```
that looks like a syntax error, did you mean to add a } before function?

rexxars replied:
Oops, thanks!

rstacruz commented on .eslintrc:
```diff
@@ -0,0 +1,20 @@
+    "no-unused-vars": 2,
+    "strict": 0
+  }
+}
```
you might want to look at https://www.npmjs.com/package/semistandard (standardjs.com). it's pretty much very close to this eslint config + more.

rexxars replied:
Thanks, will check that out

rexxars commented:
On request from @rstacruz I've changed from eslint to semistandard, which involved changing from 4 spaces to 2.

rstacruz commented:
makes a lotta sense. :+1: for me.

pwmckenna commented:
this is awesome work. api is much improved. :100: the node version doesn't seem like that big of a deal. will it actually block install of this module if you're using an old version of node or does it just warn? either way I'd say :shipit:

rexxars commented:
NPM says `this field is advisory only`, so I guess it'll just warn. Since only the tests are using 0.12 features, I decided it was kind of stupid to require it for the whole package. I refactored the tests to use async spawn and added support for 0.10 again. Merging.
